### PR TITLE
KOGITO-4175 - Kogito Image for the new JIT DMN service in Kogito Apps

### PR DESCRIPTION
**Pull request by r00ta**
JIRA: https://issues.redhat.com/browse/KOGITO-4175

This PR aims to create a new kogito image for the new kogito-apps service JIT Runner https://github.com/kiegroup/kogito-apps/tree/master/jitexecutor

Related PRs: 
kogito-runtimes: https://github.com/kiegroup/kogito-runtimes/pull/983
kogito-apps: https://github.com/kiegroup/kogito-apps/pull/597

Many thanks for submitting your Pull Request :heart:! 

Please make sure your PR meets the following requirements:

- [x] You have read the [contributors guide](README.md#contributing-to-kogito-images-repository)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Your feature/bug fix has a testcase that verifies it
- [x] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

File: modules/kogito-jit-dmn/module.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+schema_version: 1
+name: org.kie.kogito.jit-dmn
+version: "2.0.0-snapshot"
+
+artifacts:
+  - name: kogito-jit-dmn.jar
+    url: https://repository.jboss.org/nexus/content/groups/public/org/kie/kogito/jitexecutor-dmnexecutor/2.0.0-SNAPSHOT/jitexecutor-dmnexecutor-2.0.0-SNAPSHOT-runner.jar
```
Comment (r00ta) [inline modules/kogito-jit-dmn/module.yaml]: Artifact still not available on the nexus

**Comment by r00ta** [comment]
If I'm not wrong, we have to create the new repo `kogito-jit-runner` on quay.io/kiegroup before we merge this PR, otherwise the nightly build will fail

File: README.md [hunk 2]
```diff
@@ -727,6 +729,25 @@ To know what configurations this image accepts please take a look [here](kogito-
...
 to your Kogito infrastructure on a Kubernetes cluster and provide its capabilities to your Kogito applications.
 
+### Kogito JIT Runner Component Image
+
+The Kogito JIT Runner provides a tool that allows you to submit a DMN model and evaluate it on the fly with a simple request.
+
+Basic usage:
+
+```bash
+$ docker run -it quay.io/kiegroup/kogito-jit-runner:latest
```
Comment (radtriste) [inline README.md:739]: shouldn't you provide a DMN model into the docker command ?

**Comment by spolti** [comment]
@r00ta is the PR ready for review?

File: README.md (hunk 2 above)
Comment (r00ta) [reply README.md:739]: Hi @radtriste , nope, the new `jit-runner` is a service that exposes an endpoint to evaluate a dmn model given a context. 
Screenshot from 2021-01-19 23-47-47

**Comment by r00ta** [comment]
> @r00ta is the PR ready for review?

Hi @spolti , yes please

File: README.md [hunk 3]
```diff
@@ -727,6 +729,25 @@ To know what configurations this image accepts please take a look [here](kogito-
 The [Kogito Operator](https://github.com/kiegroup/kogito-cloud-operator) can be used to deploy the Kogito Trusty UI 
 to your Kogito infrastructure on a Kubernetes cluster and provide its capabilities to your Kogito applications.
 
+### Kogito JIT Runner Component Image
+
+The Kogito JIT Runner provides a tool that allows you to submit a DMN model and evaluate it on the fly with a simple request.
```
Comment (radtriste) [inline README.md]:
```suggestion
The Kogito JIT Runner provides a tool that allows you to submit a DMN model and evaluate it on the fly with a simple HTTP request.
```

File: README.md [hunk 4]
```diff
@@ -727,6 +729,25 @@ To know what configurations this image accepts please take a look [here](kogito-
...
+```bash
+$ docker run -it quay.io/kiegroup/kogito-jit-runner:latest
+```
+
+To enable debug just use this env while running this image:
+
+```bash
+docker run -it --env SCRIPT_DEBUG=true quay.io/kiegroup/kogito-jit-runner:latest
+```
+You should notice a few debug messages being printed in the system output.
```
Comment (radtriste) [inline README.md]: Maybe add here a message to look at the swagger UI for Rest API ?

**Review by radtriste — CHANGES_REQUESTED** [review]

File: README.md (hunk 4 above)
Comment (r00ta) [reply README.md]: The service does not expose a swagger ui, I enabled it locally just to take the screenshot I sent you. 

The service exposes the html page `index.html` where you can upload a model and the request form is automatically generated. Then you can evaluate it. 
Screenshot from 2021-01-20 14-00-31

I'll add a comment about it

Comment (r00ta) [reply README.md]: Let me know if it looks good now @radtriste

**Review by radtriste — APPROVED** [review]

**Comment by r00ta** [comment]
Hi @spolti , wdyt about this one?

**Comment by spolti** [comment]
@r00ta what is the target version of this changes?

File: README.md [hunk 5]
```diff
@@ -727,6 +729,25 @@ To know what configurations this image accepts please take a look [here](kogito-
 The [Kogito Operator](https://github.com/kiegroup/kogito-cloud-operator) can be used to deploy the Kogito Trusty UI 
 to your Kogito infrastructure on a Kubernetes cluster and provide its capabilities to your Kogito applications.
 
+### Kogito JIT Runner Component Image
+
+The Kogito JIT Runner provides a tool that allows you to submit a DMN model and evaluate it on the fly with a simple HTTP request.
+
+Basic usage:
```
Comment (spolti) [inline README.md:736]: could you please describe how user can send a dmn file for validation?
is it a post request, a git url, both?

File: tests/features/kogito-jit-dmn.feature [hunk 6]
```diff
@@ -0,0 +1,22 @@
+@quay.io/kiegroup/kogito-jit-runner
```
Comment (spolti) [inline tests/features/kogito-jit-dmn.feature:1]: is that possible to provide a sample dmn file and ask the image to validate/parse it?

**Review by spolti — CHANGES_REQUESTED** [review]
Nice work man, just a few comments.

File: README.md (hunk 5 above)
Comment (r00ta) [reply README.md:736]: Hi @spolti , we don't have any documentation on this new service yet. In the next days I think we will create a dedicated README in its repo `kogito-apps/jitexecutor`. If It's ok for you, I'll add here the link to that README once it is available with a dedicated small PR.

File: tests/features/kogito-jit-dmn.feature (hunk 6 above)
Comment (r00ta) [reply tests/features/kogito-jit-dmn.feature:1]: Hi @spolti , I've added a dedicated test with a small DMN model. Let me know if this looks good to you!

**Comment by r00ta** [comment]
> @r00ta what is the target version of this changes?

It would be nice to have it for the 1.2 release. I will cherry pick this if it's ok for you

**Comment by r00ta** [comment]
Here's the PR for 1.2.x https://github.com/kiegroup/kogito-images/pull/317

**Comment by radtriste** [comment]
@r00ta you mean https://github.com/kiegroup/kogito-images/pull/334 :)

**Comment by radtriste** [comment]
I guess we should have the README before merging on 1.2.x anyway. and also, we could put something in https://github.com/kiegroup/kie-docs/tree/master-kogito

**Comment by r00ta** [comment]
> I guess we should have the README before merging on 1.2.x anyway. and also, we could put something in https://github.com/kiegroup/kie-docs/tree/master-kogito

Makes sense, working on them now :+1:

**Comment by r00ta** [comment]
Jenkins retest this please

**Review by radtriste — APPROVED** [review]

**Comment by r00ta** [comment]
> I guess we should have the README before merging on 1.2.x anyway. and also, we could put something in https://github.com/kiegroup/kie-docs/tree/master-kogito

Hi @radtriste , I talked to the team about the official documentation for this service but it thinks it's too early to put it on the official documentation since it's an experimental feature (and they have other concerns). 
cc @tarilabs . 

If you agree, we can move forward with the kogito-apps README I linked

File: tests/features/kogito-jit-dmn.feature (hunk 6 above)
Comment (spolti) [reply tests/features/kogito-jit-dmn.feature:1]: cool, thanks.

File: README.md (hunk 2 above)
Comment (spolti) [reply README.md:739]: can we mention in the readme that this image exposes the swagger-ui?

File: README.md (hunk 5 above)
Comment (spolti) [reply README.md:736]: we don't need a full doc about it, just instructions on how user should use it,
like
curl .....-X POST -d .... http:....

and what user should expect as response.

**Review by spolti — CHANGES_REQUESTED** [review]

**Comment by radtriste** [comment]
> > I guess we should have the README before merging on 1.2.x anyway. and also, we could put something in https://github.com/kiegroup/kie-docs/tree/master-kogito
> 
> Hi @radtriste , I talked to the team about the official documentation for this service but it thinks it's too early to put it on the official documentation since it's an experimental feature (and they have other concerns).
> cc @tarilabs .
> 
> If you agree, we can move forward with the kogito-apps README I linked

I am good with that.

**Comment by radtriste** [comment]
FYI I create `quay.io/kiegroup/kogito-jit-runner` and `quay.io/kiegroup/kogito-jit-runner-nightly` on quay as public images so it will be ok for nightly/release pipelines
also this should be automatic once https://github.com/kiegroup/kogito-images/pull/327 will be merged

**Comment by r00ta** [comment]
Hi @spolti , the service does not expose a swagger-ui at the moment (maybe it will be added in the future). 

Since this is experimental and the endpoints/contacts might not be stable yet, I'm a bit worried to duplicate the documentation here. People might not know that they have to update the instructions here as well. 

Let me know if we should report here some instructions here and keep them up to date

**Review by spolti — APPROVED** [review]

**Pull request merged**